src/vsr/superblock.zig

```diff
@@ -883,21 +950,22 @@ pub fn SuperBlockType(comptime Storage: type) type {
                     std.mem.zeroes(vsr.Header),
                 );
             } else {
-                assert(context.caller == .checkpoint);
+                assert(!context.caller.updates_vsr_headers());
             }
 
-            if (context.caller != .view_change) {
+            if (context.caller.updates_trailers()) {
+                superblock.staging.parent_checkpoint_id = superblock.staging.checkpoint_id();
```

Comment:
I am a bit confused here. `parent_checkpoint_id` should point to parent's checkpoint_id, so I would expect.

```
superblock.staging.parent_checkpoint_id = superblock.working.checkpoint_id();
```

Reply:
`parent_checkpoint_id` was a poor naming choice, and the comment is even more misleading.

It is supposed to refer to the checkpoint which precedes the current one. But due to view changes and other (non-checkpoint) superblock updates, that does not necessarily correspond the the parent superblock.

I'll rename it to `previous_checkpoint_id`, and update the comment to:

```
/// The id of the checkpoint which precedes our current checkpoint (`vsr_state.commit_min`).
```

Reply:
Actually I think `previous_checkpoint_id` should move into the `VSRState`, to be updated by `checkpoint` and `sync_start`. That way after completing a state sync it is set to the proper value, rather than the "true" previous checkpoint.